### PR TITLE
faet(multi-threads): added support for checking multiple components

### PR DESCRIPTION
**Pull request by aberenshtein**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,54 +16,76 @@ This project is available at PyPI: [https://pypi.python.org/pypi/cachet-url-moni
...
-  latency_unit: ms
-frequency: 30
+  token: mytoken
 ```
 
-- **endpoint**, the configuration about the URL that will be monitored.
-    - **url**, the URL that is going to be monitored.
-    - **method**, the HTTP method that will be used by the monitor.
+- **endpoints**, the configuration about the URL/Urls that will be monitored.
+    - **Name**, The name of the component. Only used for readability
```
Comment (lbachar) [inline README.md]: lower case `n`

File: README.md [hunk 2]
```diff
@@ -16,54 +16,76 @@ This project is available at PyPI: [https://pypi.python.org/pypi/cachet-url-moni
...
-  public_incidents: true
-  latency_unit: ms
-frequency: 30
+  token: mytoken
 ```
 
-- **endpoint**, the configuration about the URL that will be monitored.
-    - **url**, the URL that is going to be monitored.
-    - **method**, the HTTP method that will be used by the monitor.
+- **endpoints**, the configuration about the URL/Urls that will be monitored.
```
Comment (lbachar) [inline README.md]: explain more about supporting multiple endpoints

File: cachet_url_monitor/scheduler.py [hunk 3]
```diff
@@ -87,5 +98,11 @@ def start(self):
         logging.getLogger('cachet_url_monitor.scheduler').fatal('Missing configuration file argument')
         sys.exit(1)
 
-    scheduler = Scheduler(sys.argv[1])
-    scheduler.start()
+    config_file = load(open(sys.argv[1], 'r'))
+
+    if 'api_url' and 'token' not in config_file['cachet'] or ('endpoints' not in config_file):
```
Comment (lbachar) [inline cachet_url_monitor/scheduler.py]: please keep the base configuration object for mandatory files and pass it from there to where you need
putting hard-coded fields here

File: cachet_url_monitor/scheduler.py [hunk 4]
```diff
@@ -87,5 +98,11 @@ def start(self):
...
+        logging.getLogger('cachet_url_monitor.scheduler').fatal('Missing mandatory fields')
+        sys.exit(1)
+
+    for i in range(len(config_file['endpoints'])):
```
Comment (lbachar) [inline cachet_url_monitor/scheduler.py]: Verify endpoints length is at least 1

File: cachet_url_monitor/scheduler.py (hunk 3 above)
Comment (aberenshtein) [reply cachet_url_monitor/scheduler.py]: I also don't like the idea of adding hard coded fields
So I think the best solution here is to export the validate method outside the Scheduler class.
But it has a specific check for expectations so I think it's a bit of an over kill

File: README.md (hunk 2 above)
Comment (aberenshtein) [reply README.md]: Don't you think that having an example is enough?

File: cachet_url_monitor/configuration.py [hunk 5]
```diff
@@ -136,36 +132,31 @@ def get_action(self):
...
-            for sub_key in sub_entries:
-                if sub_key not in self.data[key]:
-                    configuration_errors.append('%s.%s' % (key, sub_key))
-
-        if ('endpoint' in self.data and 'expectation' in
-            self.data['endpoint']):
-            if (not isinstance(self.data['endpoint']['expectation'], list) or
-                    (isinstance(self.data['endpoint']['expectation'], list) and
-                             len(self.data['endpoint']['expectation']) == 0)):
+        if 'expecatation' in self.endpoint:
```
Comment (lbachar) [inline cachet_url_monitor/configuration.py]:
```suggestion
        if 'expectation' in self.endpoint:
```

**Pull request merged**